feat:show transaction history for current wallet

SarveshLimaye opened the pull request:
- Added functionality to fetch wallet transaction history 

@ezequiel-rodriguez @chrisarevalodev Could you pls take a look ?

![Screenshot 2024-12-06 at 11 19 58 PM](https://github.com/user-attachments/assets/67f476f1-2aae-4a6d-8bd3-b319efc9928d)




![Screenshot 2024-12-06 at 11 20 24 PM](https://github.com/user-attachments/assets/454f2e08-9065-4525-9407-79b06bddb6f5)

ezequiel-rodriguez commented:
Hey @SarveshLimaye,
please submit your contribution in the following [form](https://forms.gle/aF9pFMfRyoygfzJWA) in order to get a reward for it.
Thank you

ezequiel-rodriguez commented:
Hey,
First of all, thanks for all the hard working you are doing and all your contributions.   

Regarding this new feature, here is some feedback:
* Alchemy key being asked each time, it is not a great UX.  My suggestion is that it can be included as a parameter in the command.   This way it can be reused each time the user call the command

*  The response from alchemy being thrown directly to the output as a json is not cool.  Check the other commands as examples.   In case for special fields or to have some flexibility if alchemy changes their response, you can add a section where you parse each pair of parameter/value and show them as they are.

SarveshLimaye commented:
@ezequiel-rodriguez Thank you so much for reviewing the changes and sharing feedback. 

I have tried to address your feedback - 

- Now, I am taking the alchemy key from the parameter, saving it in the `rootstock-wallet.json` file, and using it for further calls.
- Instead of returning a response directly from Alchemy api, I am parsing each pair of parameter/value and returning it to the user

It would be great if you could review the changes. 

Thank you!

chrisarevalodev commented:
Hey, @SarveshLimaye! Reviewed your changes and is looking very well. I just have two comments.

1. For edge case please find a way to just print one error message instead of two for the same reason
![image](https://github.com/user-attachments/assets/d5db30a2-e796-4357-98bb-3fb10b8c0b86)
2. I noticed that there's no indicator for which network are the tx's being queried from. So please add a `console.log` saying something like: `Fetching transactions on Rootstock Mainnet` or testnet, whatever the user chooses.
3. Not sure if the API makes it possible but it would be cool to show the time the tx was made.
4. Also it would be cool to have an optional parameter so the user can specify how many tx's wants returned (last 3, last 5, etc.)

SarveshLimaye commented:
@chrisarevalodev Thank you so much for reviewing the changes . I have addressed all four of your suggestions 

- Fixed edge case bug showing the same error twice 
- Added an indicator for which the tx's are being queried
- Now the time the tx was made is also visible 
- Added an optional parameter so to specify how many tx's to be returned 

It would be great if you could review the changes. I would love to hear any further suggestions.


![Screenshot 2024-12-18 at 11 39 52 PM](https://github.com/user-attachments/assets/8ff1debf-5cb6-46bd-96f7-5a367c182d92)

chrisarevalodev commented:
Tested and workgin. LGTM @ezequiel-rodriguez.
